File: server/src/main/java/com/dominicbrauer/web_wordle_tim24/model/GameSession.java
package com.dominicbrauer.web_wordle_tim24.model;

import java.util.List;

public record GameSession(
  String status,
  int guesses_used,
  String current_guess,
  boolean current_guess_valid,
  List<Guess> guesses,
  List<Integer> scores
) {}

File: web/src/lib/gameSession.ts
export interface Char {
  character: string;
  scoring: string;
}

export interface Guess {
  word: string;  
  was_correct: boolean;
  character_info: Array<Char>;
}

export interface GameSession {
  status: string;
  guesses_used: number;
  current_guess: string;
  current_guess_valid: boolean;
  guesses: Array<Guess>;
  scores: Array<number>;
}